src/UI/Components/ServiceInstanceForm/Components/FieldInput.tsx

```diff
@@ -223,9 +223,18 @@ export const FieldInput: React.FC<Props> = ({
             !isNew
           }
           type={field.inputType}
-          handleInputChange={(value, _event) =>
-            getUpdate(makePath(path, field.name), value)
-          }
+          handleInputChange={(value, _event) => {
```

Comment:
We are doing this manipulation in other places too, checking if something is parseable to JSON. I'd suggest the following and reuse it where we are doing this try catch :

```ts
const handleInputChange = (value, _event) => {
  if (field.type.includes("dict")) {
    getUpdate(makePath(path, field.name), tryParseJSON(value));
  } else {
    getUpdate(makePath(path, field.name), value);
  }
};

const tryParseJSON = (value) => {
  try {
    return JSON.parse(value);
  } catch (_error) {
    // If the value is not a valid JSON string, return the original value
    return value;
  }
};
```